### ISSUE TITLE
help: download video from "(elearn|classes).brilliantpala.org"

### ISSUE THREAD

**Issue by asdfghjkl87755** [pzhlkj6612/streamlink-eplus_jp-object_storage#2]
Can you please tell me how to download this video

here is the link

https://d1j3vi2u94ebt0.cloudfront.net/institute/brilliantpalalms/courses/neet-2023/videos/transcoded/ddeb0f01b521497b9106ba3c83d37bdf/720p/video.m3u8

**Comment by pzhlkj6612** [comment]
Hi.

```console
$ curl "https://d1j3vi2u94ebt0.cloudfront.net/institute/brilliantpalalms/courses/neet-2023/videos/transcoded/ddeb0f01b521497b9106ba3c83d37bdf/720p/video.m3u8"
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:18
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-KEY:METHOD=AES-128,URI="https://elearn.brilliantpala.org/api/v2.4/encryption_key/c0c89660d07143debac34cd41a8e80a3/",IV=0x00000000000000000000000000000000
#EXTINF:11.280000,
video_0.ts
...
#EXTINF:7.800000,
video_850.ts
#EXT-X-ENDLIST
...
```

Please note the "`#EXT-X-KEY:METHOD=AES-128...`" line. This line means that the video file was encrypted, so this playlist cannot be played by VLC directly, even if you download all "*.ts" files.  [Brilliant Study Centre Pala](https://elearn.brilliantpala.org/) looks like an online education website for a limited range of registered users, thus the AES key may be protected by cookies. Unless I log in to the website, I cannot get the key. So, I have to say sorry, I'm not able to help you download the playlist.

<br>

It does not mean that there is no way to download and decrypt the video. Have a look at [ffmpeg - hey i need to download my coaching classes but they are using hls aes 128 bit encrypted streams. i downloaded m3u8 and all .ts files and found enc key - Stack Overflow](https://stackoverflow.com/q/75782081) and please note this sentence:

> encryption key from preview tab I found : Ø®¢Ù¦Aä [u request url of the key :

Do you know what the "preview tab" is?

- Chrome: [Inspect a resource's details § Inspect network activity - Chrome Developers](https://developer.chrome.com/docs/devtools/network/#details);  
- Firefox: [Response tab § Network request details — Firefox Source Docs documentation](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_details/index.html#response-tab).

You can watch your online courses with DevTools open. The web browser will request the playlist, the key and video files and you can see all network activities in the Network activity tab. Now you need to find the request about the key (by searching "`https://elearn.brilliantpala.org/api/v2.4/encryption_key/c0c89660d07143debac34cd41a8e80a3/`"). After finding the request:

- You can directly download the key by double-clicking the request. Your browser may ask you the save location.
- If you happen to have a Linux, you can convert the request to a cURL commandline and download the key with cURL:
  - Chrome: [Copy one or more requests to the clipboard § Network features reference - Chrome Developers](https://developer.chrome.com/docs/devtools/network/reference/#copy);
  - Firefox: [Context menu § Network request list — Firefox Source Docs documentation](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_list/index.html#context-menu).
- If you happen to know something about [HTTP Cookies](https://en.wikipedia.org/wiki/HTTP_cookie), there is a nice tool can help you download, decrypt and merge all video clips: [selsta/hlsdl](https://github.com/selsta/hlsdl): C program to download VoD HLS (.m3u8) files.

You need to decrypt video files by you hand for the first two methods. Please note that the content of the AES key is not printable, so the asker on StackOverflow had written the un-readable "Ø®¢Ù¦Aä [u" line in that question. Have a look at [encryption - How to decrypt AES-128 encrypted m3u8 video files? - Stack Overflow](https://stackoverflow.com/q/16132088) for decrypting video file by openssl.

<br>

Good luck.

**Comment by asdfghjkl87755** [comment]
Username - 292870
Password - 292870

Please can you check it...

video link - 
https://elearn.brilliantpala.org/courses/98/contents/36683/

**Comment by pzhlkj6612** [comment]
@asdfghjkl87755 ,

Download FFmpeg from https://github.com/yt-dlp/FFmpeg-Builds. Make sure the directory of ffmpeg.exe is in your "`PATH`". In CMD:

```console
> ffmpeg -version
ffmpeg version...

```

Install my fork of yt-dlp. In CMD:

```console
> pip install https://github.com/pzhlkj6612/yt-dlp/archive/8a4b9767563d8069b13110a487655d653f76b12e.zip
Collecting https://github.com/...

```

Then, download the video. In CMD:

```console
> yt-dlp --username "<YOUR USERNAME>" --password "<YOUR PASSWORD>" "https://elearn.brilliantpala.org/courses/98/contents/36683/"
[Brilliantpala:CourseContent] Downloading login page
[Brilliantpala:CourseContent] Logging in
...

```

**Comment by asdfghjkl87755** [comment]
Hi sir 
The above method is working perfectly to download video from https://elearn.brilliantpala.org

But I am not able to download video from 

https://classes.brilliantpala.org/

video link : https://classes.brilliantpala.org/courses/416/contents/25445/

Username 256436
Password 256436

Please give a method to download from https://classes.brilliantpala.org

**Comment by pzhlkj6612** [comment]
> Please give a method to download from https://classes.brilliantpala.org

```
pip install https://github.com/pzhlkj6612/yt-dlp/archive/0e74f985b8d3d8471ef451417599829f75d6dc52.zip
```

Give it a try.

**Comment by pzhlkj6612** [comment]
> > Please give a method to download from https://classes.brilliantpala.org
> 
> ```
> pip install https://github.com/pzhlkj6612/yt-dlp/archive/0e74f985b8d3d8471ef451417599829f75d6dc52.zip
> ```
> 
> Give it a try.

The above command may not work; use this one to overwrite the existing yt-dlp:

```bash
pip install --force-reinstall https://github.com/pzhlkj6612/yt-dlp/archive/0e74f985b8d3d8471ef451417599829f75d6dc52.zip

```

**Comment by asdfghjkl87755** [comment]
the method worked vey well .Thanks so much for helping me

**Comment by pzhlkj6612** [comment]
@asdfghjkl87755 , if there is no more online class urls you need to download, please close this issue.

**Comment by pzhlkj6612** [comment]
Hi @asdfghjkl87755 , the [official repository of yt-dlp](https://github.com/yt-dlp/yt-dlp) has accepted my code and released a new version [2023.09.24](https://github.com/yt-dlp/yt-dlp/releases/tag/2023.09.24). Now you should use the original yt-dlp instead of my fork:

```console
> pip install --force-reinstall yt-dlp
Collecting yt-dlp
  Downloading yt_dlp-2023.9.24-py2.py3-none-any.whl (3.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 2.5 MB/s eta 0:00:00
Collecting mutagen (from yt-dlp)
  Downloading mutagen-1.47.0-py3-none-any.whl (194 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 194.4/194.4 kB ? eta 0:00:00
Collecting pycryptodomex (from yt-dlp)
  Downloading pycryptodomex-3.19.0-cp35-abi3-win_amd64.whl (1.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 13.9 MB/s eta 0:00:00
Collecting websockets (from yt-dlp)
  Downloading websockets-11.0.3-cp311-cp311-win_amd64.whl (124 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 124.7/124.7 kB 7.6 MB/s eta 0:00:00
Collecting certifi (from yt-dlp)
  Downloading certifi-2023.7.22-py3-none-any.whl (158 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.3/158.3 kB 9.3 MB/s eta 0:00:00
Collecting brotli (from yt-dlp)
  Downloading Brotli-1.1.0-cp311-cp311-win_amd64.whl (357 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 357.3/357.3 kB 21.7 MB/s eta 0:00:00
Installing collected packages: brotli, websockets, pycryptodomex, mutagen, certifi, yt-dlp
Successfully installed brotli-1.1.0 certifi-2023.7.22 mutagen-1.47.0 pycryptodomex-3.19.0 websockets-11.0.3 yt-dlp-2023.9.24

> yt-dlp --version
2023.09.24

```

**Comment by pzhlkj6612** [comment]
@asdfghjkl87755 , don't forget to specify your username and password:

```console
> yt-dlp --username "<YOUR USERNAME>" --password "<YOUR PASSWORD>" "<URL>"

```
